set JULIA_TEST_MAXRSS_MB=600 on appveyor

tkelman opened the pull request:
in anticipation of appveyor potentially moving to 4 cores again

don't merge this until we start seeing builds happen on more cores

tkelman committed:
set JULIA_TEST_MAXRSS_MB=600 on appveyor

in anticipation of appveyor potentially moving to 4 cores again

JeffBezanson commented:
+1 Can we do this on travis too? I think this might help regardless of the number of cores. I suspect there are some slight super-linearities as code accumulates in the cache, and restarting more might make the tests faster overall. For example here's a small local experiment:

```
jeff@gurren:~/src/julia/test$ JULIA_CPU_CORES=1 ../julia runtests.jl misc
     * misc                  in  54.57 seconds, maxrss  345.02 MB

jeff@gurren:~/src/julia/test$ JULIA_CPU_CORES=1 ../julia runtests.jl core misc
     * core                  in  14.95 seconds, maxrss  212.13 MB
     * misc                  in  69.01 seconds, maxrss  393.70 MB
```

I picked `misc` since it seems to have one of the largest slowdowns after llvm 3.7.

tkelman commented:
Worth trying I suppose. The renumbering of restarted workers can make failures/freezes a little tough to follow which is why I'm not a huge fan of this as currently implemented.

> I suspect there are some slight super-linearities as code accumulates in the cache

That's definitely another bullet to add to #14743...

tkelman committed:
also set JULIA_TEST_MAXRSS_MB=600 on travis

and split some long lines
(hopefully the yml parsing gods don't get angered by this)

tkelman commented:
Yeah, this does save a few minutes. Bit better throughput is worth the slightly more confusing bookkeeping.
